src/ApiManagement/ApiManagement.ServiceManagement/Commands/AddAzureApiManagementApiToGateway.cs

```diff
@@ -0,0 +1,69 @@
+﻿//  
+// Copyright (c) Microsoft.  All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//    http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+namespace Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands
+{
+    using Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Models;
+    using Microsoft.Azure.Management.ApiManagement.Models;
+    using System;
+    using System.Management.Automation;
+
+    [Cmdlet("Add", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApiManagementApiToGateway", SupportsShouldProcess = true)]
+    [OutputType(typeof(bool))]
+    public class AddAzureApiManagementApiToGateway : AzureApiManagementCmdletBase
+    {
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Instance of PsApiManagementContext. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public PsApiManagementContext Context { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Identifier of existing gateway. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public String GatewayId { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Identifier of existing API. This parameter is required.")]
+        [ValidateNotNullOrEmpty]
+        public String ApiId { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "Provisioning State (Created). This parameter is optional.")]
+        public PsApiManagementGatewayApiProvisioningState? ProvisioningState { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
+            HelpMessage = "If specified will write true in case operation succeeds. This parameter is optional. Default value is false.")]
+        public SwitchParameter PassThru { get; set; }
+
+        public override void ExecuteApiManagementCmdlet()
+        {
+            Client.ApiAddToGateway(Context, GatewayId, ApiId, ProvisioningState);
+
+            if (PassThru)
+            {
+                WriteObject(true);
```

Comment:
Why not return the Api Contract:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2019-12-01/apimgateways.json#L930

Reply:
@VeryEarly , for consistency, the same way as AddAzureApiManagementApiToProduct.cs   
Also, this operation doesn't change ApiContract or GatewayContract, it just creates an association between them. I think no need to return any contract here.